apps/aecore/src/aec_db_gc.erl

```diff
@@ -0,0 +1,211 @@
+-module(aec_db_gc).
+
+-behaviour(gen_statem).
+
+%% API
+-export([start_link/0,
+         start_link/3,
+         maybe_garbage_collect/0,
+         stop/0]).
+
+%% gen_statem callbacks
+-export([callback_mode/0, init/1, terminate/3, code_change/4]).
+-export([handle_event/4]).
+
+-record(data,
+        {enabled  :: boolean(),                         % do we garbage collect?
+         interval :: non_neg_integer(),                 % how often (every `interval` blocks) GC runs
+         history  :: non_neg_integer(),                 % how many block state back from top to keep
+         synced   :: boolean(),                         % we only run GC if chain is synced
+         height   :: undefined | non_neg_integer(),     % latest height of MPT hashes stored in tab
+         hashes   :: undefined | pid() | reference()}). % hashes tab (or process filling the tab 1st time)
+
+-include_lib("aecore/include/aec_db.hrl").
+
+-define(DEFAULT_INTERVAL, 50000).
+-define(DEFAULT_HISTORY, 500).
+
+-define(TIMED(Expr), timer:tc(fun () -> Expr end)).
+-define(LOG(Fmt), lager:info(Fmt, [])).         % io:format(Fmt"~n")
+-define(LOG(Fmt, Args), lager:info(Fmt, Args)). % io:format(Fmt"~n", Args)
+
+%% %%%===================================================================
+%% %%% API
+%% %%%===================================================================
+
+start_link() ->
+    #{enabled := Enabled, interval := Interval, history := History} = config(),
```

Comment:
Now, there isn't currently any API for changing config params during run-time (except during testing), but perhaps there should still be a one-line comment noting that the implementation expects the node to be restarted in order to change the GC parameters? Specifically, the fsm never gets out of `synced = false` unless a `chain_sync_done` event is received, which only happens if subscription is enabled, which only happens in `init/1` if `enabled = true` from the start.

Reply:
Yes, that's correct. There doesn't seem to be practical utility to support runtime reconfiguration, adding comment there.